### PR TITLE
Fix patient equality and hashcode bug and update storage loading related test methods (Order: 1)

### LINKED ISSUE #141: Fix patient equality bug
Patient#equals() and Patient#hashCode() does not include medicalNoteList

### PR DESCRIPTION
**Pull request by wang-jun-hao**
Fixes #141 
Update test methods where the expected behaviour is that MediBook is loaded from storage with all patients holding empty medical note list (as per program specification).

### CODE REVIEW

File: src/test/java/seedu/medibook/storage/JsonAdaptedPatientTest.java [hunk 1]
```diff
@@ -49,11 +52,17 @@
     private static final List<JsonAdaptedTag> VALID_TAGS = BENSON.getTags().stream()
             .map(JsonAdaptedTag::new)
             .collect(Collectors.toList());
+    // medical note list data of patient should not be loaded at point of program start-up, hence expected
+    // behaviour is that all these patients have an empty medical note list instead of the one initialised and loaded
+    // in TypicalPatients.java for other testing purposes
```
Comment (foojingyi) [inline src/test/java/seedu/medibook/storage/JsonAdaptedPatientTest.java]: May look better to properly punctuate this comment?

File: src/test/java/seedu/medibook/storage/JsonMediBookStorageTest.java [hunk 2]
```diff
@@ -64,25 +68,35 @@ public void readMediBook_invalidAndValidPatientMediBook_throwDataConversionExcep
     public void readAndSaveMediBook_allInOrder_success() throws Exception {
         Path filePath = testFolder.resolve("TempMediBook.json");
         MediBook original = getTypicalMediBook();
+        // this is the expected MediBook when program loads.
```
Comment (foojingyi) [inline src/test/java/seedu/medibook/storage/JsonMediBookStorageTest.java]: Maybe capitalise this comment?

**Review by foojingyi — APPROVED** [review]
LGTM, just a few questions about comments. While looking at the changed class I also wonder if "zeroBasedIndex" would be a more readable variable name than "indexZeroBased"?

File: src/test/java/seedu/medibook/storage/JsonMediBookStorageTest.java (hunk 2 above)
Comment (wang-jun-hao) [reply src/test/java/seedu/medibook/storage/JsonMediBookStorageTest.java]: Ok thanks I have fixed this

File: src/test/java/seedu/medibook/storage/JsonAdaptedPatientTest.java (hunk 1 above)
Comment (wang-jun-hao) [reply src/test/java/seedu/medibook/storage/JsonAdaptedPatientTest.java]: Ok thanks I have fixed this

**Pull request merged**